src/Parser/HParser.y

```diff
@@ -0,0 +1,299 @@
+      | match Expr with ListE1(Arm)            { withPos2 $1 $3 $ Match $2 $4 }
+      | Atom                                   { $1 }
+
+Atom :: { Expr Parsed }
+     : Var                                    { withPos1 $1 (VarRef (getL $1)) }
+     | Con                                    { withPos1 $1 (VarRef (getL $1)) }
+     | Lit                                    { withPos1 $1 (Literal (getL $1)) }
+     | hole                                   { withPos1 $1 (Hole (Name (getHole $1))) }
+     | begin List1(Expr, ';') end             { withPos2 $1 $3 $ Begin $2 }
+     | '(' List(Expr, ',') ')'                { withPos2 $1 $3 $ tupleExpr $2 }
+     | '{' Rows('=', Expr) '}'                { withPos2 $1 $3 $ Record $2 }
+     | '{' Expr with Rows('=',Expr) '}'       { withPos2 $1 $5 $ RecordExt $2 $4 }
+
+     | '(' access ')'                         { withPos2 $1 $3 $ AccessSection (getIdent $2) }
+     | Atom access                            { withPos2 $1 $2 $ Access $1 (getIdent $2) }
+     | '(' Operator ')'                       { withPos2 $1 $3 $ BothSection $2 }
+     | '(' Expr Operator ')'               { withPos2 $1 $4 $ RightSection $2 $3 }
+     | '(' Operator Expr ')'               { withPos2 $1 $4 $ LeftSection $2 $3 }
+
+Operator :: { Expr Parsed }
+         : '**'                               { withPos1 $1 $ varE "**" }
+         | '*'                                { withPos1 $1 $ varE "*" }
+         | '/'                                { withPos1 $1 $ varE "/" }
+         | '+'                                { withPos1 $1 $ varE "+" }
+         | '-'                                { withPos1 $1 $ varE "-" }
+         | '^'                                { withPos1 $1 $ varE "^" }
+         | '<'                                { withPos1 $1 $ varE "<" }
+         | '>'                                { withPos1 $1 $ varE ">" }
+         | '<='                               { withPos1 $1 $ varE "<=" }
+         | '>='                               { withPos1 $1 $ varE ">=" }
+         | '<>'                               { withPos1 $1 $ varE "<>" }
+         | '=='                               { withPos1 $1 $ varE "==" }
+         | '&&'                               { withPos1 $1 $ varE "&&" }
+         | '||'                               { withPos1 $1 $ varE "||" }
+
+Var :: { Located (Var Parsed) }
+    : ident { lPos1 $1 $ Name (getIdent $1) }
+
+Con :: { Located (Var Parsed) }
+    : conid { lPos1 $1 $ Name (getIdent $1) }
+
+TyVar :: { Var Parsed }
+      : tyvar { Name (getIdent $1) }
+
+BindGroup :: { [(Var Parsed, Expr Parsed, Ann Parsed)] }
+          : Binding                           { [$1] }
+          | BindGroup and Binding             { $3 : $1 }
+
+Binding :: { (Var Parsed, Expr Parsed, Ann Parsed) }
+        : Var ListE(ArgP) '=' Expr            { (getL $1, foldr (\x y -> withPos2 x $4 (Fun x y)) $4 $2, withPos2 $1 $4 id) }
```

Comment:
Hmm, seems like you forgot the optional type annotation here:
```
| Var ListE(ArgP) ':' Type '=' Expr = { Ascription {- the stuff -} {- the type -} }
```

Reply:
Done.